packages/react-query/src/__tests__/useQueries.test.tsx

```diff
@@ -145,7 +145,7 @@ describe('useQueries', () => {
             return (i + 1) * count * 2
           },
         })),
-      })
+      }) as QueryObserverResult<unknown, Error>[]
```

Comment:
why is this change necessary?

Reply:
we needed to infer the types here because the type of queries here is unknown altho the states is waiting a type Error as a default type

Reply:
the Problem seems to be that result is inferred as:

```
const result: UseQueryResult<number, unknown>[]
```

while I think it should be:

```
const result: UseQueryResult<number, Error>[]
```

If we just bypass the issue here in the test, users might suffer from the same issue as well, so I think we should strive towards fixing the issue.

I tried to tinker a bit with the `useQueries` types but with no success. I've reverted some of the type assertions (https://github.com/TanStack/query/pull/4706/commits/9e6368b575ae70f52235eeb9412e85fdeb538743) to have the state we currently have on `main`. I was hoping that @artysidorenko could maybe have a look? I think he knows what to do to make types default to `Error` instead of `unknown`. I think all the issue left are with `useQueries`.

Reply:
yes you're totally right, we need to fix instead of bypassing the error, I'll to find a fix for the useQueries, but if the guy you mentioned can help that would be great

Reply:
Hi folks - I think [this](https://github.com/artysidorenko/react-query/commit/3fddbffec6f5301c92e99adff721ebc11d426927) should do it for the TError type - it was just missing the defaulting on dynamic arrays.

Also saw a small type complaint in the function body of `useQueries` because the queryClient.defaultQueryOptions method [didn't have any default type params](https://github.com/artysidorenko/react-query/commit/35b3aaa855e138341afb871999487766bd0affd2)